diff -up chromium-136.0.7103.33/net/filter/brotli_source_stream.cc.system-brotli chromium-136.0.7103.33/net/filter/brotli_source_stream.cc
--- chromium-136.0.7103.33/net/filter/brotli_source_stream.cc.system-brotli	2025-04-15 23:18:07.000000000 +0200
+++ chromium-136.0.7103.33/net/filter/brotli_source_stream.cc	2025-04-21 11:12:15.771632937 +0200
@@ -17,8 +17,8 @@
 #include "base/metrics/histogram_macros.h"
 #include "net/base/io_buffer.h"
 #include "net/filter/source_stream_type.h"
-#include "third_party/brotli/include/brotli/decode.h"
-#include "third_party/brotli/include/brotli/shared_dictionary.h"
+#include <brotli/decode.h>
+#include <brotli/shared_dictionary.h>
 
 namespace net {
 
diff -up chromium-136.0.7103.33/net/ssl/cert_compression.cc.system-brotli chromium-136.0.7103.33/net/ssl/cert_compression.cc
--- chromium-136.0.7103.33/net/ssl/cert_compression.cc.system-brotli	2025-04-15 23:18:07.000000000 +0200
+++ chromium-136.0.7103.33/net/ssl/cert_compression.cc	2025-04-21 11:12:15.775633048 +0200
@@ -9,7 +9,7 @@
 #include "third_party/boringssl/src/include/openssl/ssl.h"
 
 #if !defined(NET_DISABLE_BROTLI)
-#include "third_party/brotli/include/brotli/decode.h"
+#include <brotli/decode.h>
 #endif
 
 namespace net {
diff -up chromium-136.0.7103.33/ui/base/resource/resource_bundle.cc.system-brotli chromium-136.0.7103.33/ui/base/resource/resource_bundle.cc
--- chromium-136.0.7103.33/ui/base/resource/resource_bundle.cc.system-brotli	2025-04-15 23:18:07.000000000 +0200
+++ chromium-136.0.7103.33/ui/base/resource/resource_bundle.cc	2025-04-21 11:19:08.136223039 +0200
@@ -39,7 +39,7 @@
 #include "build/build_config.h"
 #include "net/filter/gzip_header.h"
 #include "skia/ext/image_operations.h"
-#include "third_party/brotli/include/brotli/decode.h"
+#include <brotli/decode.h>
 #include "third_party/skia/include/codec/SkPngDecoder.h"
 #include "third_party/skia/include/core/SkBitmap.h"
 #include "third_party/skia/include/core/SkColor.h"
