diff -up chromium-140.0.7339.80/build/config/compiler/BUILD.gn.me chromium-140.0.7339.80/build/config/compiler/BUILD.gn
--- chromium-140.0.7339.80/build/config/compiler/BUILD.gn.me	2025-09-03 15:51:28.599633500 +0200
+++ chromium-140.0.7339.80/build/config/compiler/BUILD.gn	2025-09-03 15:52:46.285444744 +0200
@@ -2134,9 +2134,6 @@ config("default_warnings") {
 
       # TODO(crbug.com/40284799): Fix and re-enable.
       "-Wno-thread-safety-reference-return",
-
-      # TODO(crbug.com/376641662): Fix and re-enable.
-      "-Wno-nontrivial-memcall",
     ]
 
     cflags_cc += [
