diff --git a/ui/gfx/linux/gbm_wrapper.cc b/ui/gfx/linux/gbm_wrapper.cc
index e46fbaca92..5ea82a657f 100644
--- a/ui/gfx/linux/gbm_wrapper.cc
+++ b/ui/gfx/linux/gbm_wrapper.cc
@@ -421,6 +421,9 @@ class Device final : public ui::GbmDevice {
       fd_data.fds[i] = handle.planes[i < handle.planes.size() ? i : 0].fd.get();
       fd_data.strides[i] = handle.planes[i].stride;
       fd_data.offsets[i] = handle.planes[i].offset;
+      LOG(ERROR) << "!!! gbm_bo_import i " << i;
+      LOG(ERROR) << "!!! gbm_bo_import stride " << handle.planes[i].stride;
+      LOG(ERROR) << "!!! gbm_bo_import offset " << handle.planes[i].offset;
     }
 
     // The fd passed to gbm_bo_import is not ref-counted and need to be
@@ -429,11 +432,36 @@ class Device final : public ui::GbmDevice {
                                       &fd_data, gbm_flags);
     if (!bo) {
       if (gbm_flags & GBM_BO_USE_SCANOUT) {
-        gbm_flags &= ~GBM_BO_USE_SCANOUT;
         bo = gbm_bo_import(device_.get(), GBM_BO_IMPORT_FD_MODIFIER, &fd_data,
+                           gbm_flags & ~GBM_BO_USE_SCANOUT);
+      }
+
+      if (!bo && handle.modifier == gfx::NativePixmapHandle::kNoModifier) {
+        LOG(ERROR) << "!!! testing GBM_BO_IMPORT_FD";
+        struct gbm_import_fd_data new_fd_data;
+        new_fd_data.fd = fd_data.fds[0];
+        new_fd_data.width = fd_data.width;
+        new_fd_data.height = fd_data.height;
+        new_fd_data.stride = fd_data.strides[0];
+        new_fd_data.format = fd_data.format;
+
+        bo = gbm_bo_import(device_.get(), GBM_BO_IMPORT_FD, &new_fd_data,
                            gbm_flags);
       }
 
+      if (!bo && handle.modifier == gfx::NativePixmapHandle::kNoModifier) {
+        LOG(ERROR) << "!!! testing GBM_BO_IMPORT_FD without GBM_BO_USE_SCANOUT";
+        struct gbm_import_fd_data new_fd_data;
+        new_fd_data.fd = fd_data.fds[0];
+        new_fd_data.width = fd_data.width;
+        new_fd_data.height = fd_data.height;
+        new_fd_data.stride = fd_data.strides[0];
+        new_fd_data.format = fd_data.format;
+
+        bo = gbm_bo_import(device_.get(), GBM_BO_IMPORT_FD, &new_fd_data,
+                           gbm_flags & ~GBM_BO_USE_SCANOUT);
+      }
+
       if (!bo) {
         LOG(ERROR) << "nullptr returned from gbm_bo_import";
         return nullptr;
